tools/topology/sof/pipe-kfbm-capture.m4

```diff
@@ -26,7 +52,7 @@ define(`N_STS', `Sound Trigger Capture '$1)
 W_PCM_CAPTURE(PCM_ID, Sound Trigger Capture, 0, 2, 2)
 
 # "KPBM" has 2 source and 2 sink periods
-W_KPBM(0, PIPELINE_FORMAT, 2, 2, PIPELINE_ID)
+W_KPBM(0, PIPELINE_FORMAT, 2, 2, PIPELINE_ID, LIST(`             ', "KPB"))
```

Comment:
What is this list for an what do you mean by *"KPBM" has 2 source and 2 sink periods*. KPB has 2 sinks, namely selector and host but as for sources I thought we only have one? We also use only one period, so please clarify it.

Reply:
> What is this list for an what do you mean by _"KPBM" has 2 source and 2 sink periods_. KPB has

the list is alsa mixer(kcontrol) list we will create for this kpbm widget in driver side.

> 2 sinks, namely selector and host but as for sources I thought we only have one? We also use only one period, so please clarify it.

Here it means we have both 2 periods for source(connected with dmic) and sink(connected with host).

Reply:
@keyonjie, could you please say what role will this kcontrol play on FW side?
You send some initial values like *bytes "0x53,0x4f,...* how can one know what they initialize, for what purpose and why is it needed in firmware?

Reply:
As stated in the comment message(and also I talked to you offline yesterday), it is kpb initial parameters, aligned with struct sof_kpb_config, the first 32Bytes is header as what we have already used for EQ.